### ISSUE TITLE
[Bug Report] Incorrect A2C implementation

### ISSUE THREAD

**Issue by LeoUschi232** [Farama-Foundation/Gymnasium#1494]
### Describe the bug

In A2C implementation.
The critical bug is that you use advantages as the critic target without detaching it, so the critic loss backpropagates through the GAE computation itself (i.e., through value_preds[t] and value_preds[t+1] inside td_error and the recursion). Concretely, advantages is computed from value_preds with gradients enabled, then critic_loss = advantages.pow(2).mean() sends gradients back through that whole bootstrapped chain.
This is not the intended value-function regression (you want to regress V to a fixed target).
Correct:
```
    def get_losses(
            self,
            rewards: torch.Tensor,
            action_log_probs: torch.Tensor,
            value_preds: torch.Tensor,
            entropy: torch.Tensor,
            masks: torch.Tensor,
            gamma: float,
            lam: float,
            ent_coef: float,
            device: torch.device,
    ) -> tuple[torch.Tensor, torch.Tensor]:
        T = len(rewards)
        advantages = torch.zeros(T, self.n_envs, device=device)
        # compute the advantages using GAE
        gae = 0.0
        for t in reversed(range(T - 1)):
            td_error = (
                    rewards[t] + gamma * masks[t] * value_preds[t + 1] - value_preds[t]
            )
            gae = (td_error + gamma * lam * masks[t] * gae).detach()
            advantages[t] = gae
        # calculate the loss of the minibatch for actor and critic
        critic_loss = (value_preds - (advantages + value_preds.detach())).pow(2).mean()
        # give a bonus for higher entropy to encourage exploration
        actor_loss = (
                -(advantages.detach() * action_log_probs).mean() - ent_coef * entropy.mean()
        )
        return critic_loss, actor_loss
```

### Code example

```shell

```

### Additional context

Website: https://gymnasium.farama.org/tutorials/training_agents/vector_a2c/

### Checklist

- [ ] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/Gymnasium/issues) in the repo

**Comment by LeoUschi232** [comment]
Actually if I'm gonna fix this myself, I would prefer to also add a final bootstrap state value on state values.
Almost every tensor in the rollouts should have episode length T but state values should have T+1 where s_{T+1} is the bootstrap value and defaults to 0 on terminal states. This would also allow computing of T advantages instead of T-1 advantages as each advantage A_t uses V_t and V_{t+1} in this case you would need V_{T+1} for advantage A_T. Do you want me to fix both issues in the pull request? @pseudo-rnd-thoughts
